src/main/service/listManipulation/crownFirmListService.ts

```diff
@@ -0,0 +1,116 @@
+import moment from 'moment-timezone';
+import {DataManipulationService} from '../dataManipulationService';
+import {DateTimeHelper} from '../../helpers/dateTimeHelper';
+import {CrimeListsService} from './CrimeListsService';
+
+const dataManipulationService = new DataManipulationService();
+const dailyListService = new CrimeListsService();
+const dateTimeHelper = new DateTimeHelper();
+
+export class CrownFirmListService {
+  public timeZone = 'Europe/London';
+
+  /**
+   * Builds list of courthouses, does general formatting to make it appear nice, then is passed to second function which
+   * deals with the splitting it by various dimensions (i.e. courthouse -> day -> courtRoom -> allocated/unallocated.
+   * @param firmList
+   * @param language
+   * @param languageFile
+   */
+  public splitOutFirmListData(firmList: string, language: string, languageFile: string) {
+    const rows = [];
+    const firmListData = JSON.parse(firmList);
+    firmListData['courtLists'].forEach(courtList => {
+      const courtName = courtList['courtHouse']['courtHouseName'];
+      courtList['courtHouse']['courtRoom'].forEach(courtRoom => {
+        courtRoom['session'].forEach(session => {
+          session['sittings'].forEach(sitting => {
+            const sittingDate = moment.utc(sitting['sittingStart']).tz(this.timeZone).format('dddd DD MMMM YYYY');
+            sitting['formattedDuration'] = dateTimeHelper.formatDuration(sitting['durationAsDays'] as number, sitting['durationAsHours'] as number,
+              sitting['durationAsMinutes'] as number, language, languageFile);
+            sitting['hearing'].forEach(hearing => {
+              dailyListService.findLinkedCasesInformation(hearing);
+              hearing['case'].forEach(thisCase => {
+                const formattedName = hearing['defendant'].split(/\s+/);
+                const row = {
+                  courtName: courtName,
+                  sittingDate: sittingDate,
+                  sittingTime: dataManipulationService.publicationTimeInBst(sitting['sittingStart']),
+                  courtRoom: courtRoom['courtRoomName'],
+                  joh: session['formattedJudiciaries'],
```

Comment:
formattedJudiciaries is not working as expected. If there is Judiciaries section in SITTING but not in SESSION, it will ignore the SITTING Judiciaries data. This is bug which I fixed in Mags Standard list. 
You can fix it:
In SittingLoop, do something like
`session['sittings'].forEach(sitting => {
            const judiciary = dataManipulationService.findAndManipulateJudiciary(sitting);
            if (judiciary !== '') {
              session['formattedJudiciaries'] = judiciary;
            }`

Reply:
> Formatting of table is not align with mockup. We use to have horizontal scroll bar for the table which is missing in this list.
> 
> ![error](https://user-images.githubusercontent.com/84805836/201683918-7b341855-4f4e-4968-8222-a4824ce60df4.jpg)

Sorted :)